detectors/gcp/gce.go

```diff
@@ -73,3 +79,36 @@ func (d *Detector) GCEAvailabilityZoneAndRegion() (string, string, error) {
 	}
 	return zone, strings.Join(splitZone[0:2], "-"), nil
 }
+
+type ManagedInstanceGroup struct {
+	Name     string
+	Location string
+	Type     LocationType
+}
+
+var createdByMIGRE = regexp.MustCompile(`^projects/[^/]+/(zones|regions)/([^/]+)/instanceGroupManagers/([^/]+)$`)
+
+func (d *Detector) GCEManagedInstanceGroup() (ManagedInstanceGroup, error) {
+	createdBy, err := d.metadata.Get(createdByMetadataAttr)
+	if _, ok := err.(metadata.NotDefinedError); ok {
+		return ManagedInstanceGroup{}, nil
+	} else if err != nil {
+		return ManagedInstanceGroup{}, err
+	}
+	matches := createdByMIGRE.FindStringSubmatch(createdBy)
+	if matches == nil {
+		return ManagedInstanceGroup{}, nil
```

Comment:
Should we return an error here? Seems like if it is in the wrong format we should let the user know, right?

Reply:
The key is just `created-by`; my assumption is that other tools will put their own strings indicating that they created the VM, and that's not an error, it just means the VM wasn't created by a MIG.

Reply:
please add a comment :)